packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -100,6 +116,15 @@ class ImageInfo {
   /// the image.
   final ui.Image image;
 
+  /// If true, the [image] will be disposed when this object is disposed.
+  ///
+  /// If false, the object will not dispatch its lifecicle events to [FlutterMemoryAllocations].
+  ///
+  /// This is used to:
+  /// * avoid disposing the [image] when it is owned by other object (when false)
+  /// * transfer the [image] ownership to this object (when true)
+  final bool shouldDisposeImage;
```

Comment:
This sounds very dubious to me. Why do we have a disposable that is sometimes configured to not be disposed? This sounds like a foot gun. Also, the docs don't really help much in understanding when you should set this to true or false.

/cc @dnfield who maybe has more context on this from back when he added disposal support to images.

Reply:
De-facto rule for disposing is: object should dispose a member if it created it.

It is different for ImageInfo.image:
1. Sometimes image is cloned
2. Sometimes image it is passed from other object
3. Historically image was always disposed, independently on creation story

There is mess in tests around disposing for images. So, there is no way to see if the leak is test-only or real.
In order to clean it up, I need some consistent meaningful vision about how it should be.

So, by this change I want to make it possible for ImageInfo creator to decide who will now be responsible or disposal of the image, i.e. if creator wants to transfer the image ownership to ImageInfo or not.

Does it help?

Reply:
I agree that ImageInfo has some difficult to reason about behavior.

Rather than encoding that behavior in this difficult to reason about parameter, can we somehow clean it up and unify it?